src/emulate.c

```diff
@@ -582,14 +628,14 @@ static void block_translate(riscv_t *rv, block_map_t *map, block_t *block)
 
         /* decode the instruction */
         if (!rv_decode(ir, insn)) {
-            rv->compressed = (ir->insn_len == INSN_16);
+            rv->compressed = ((insn & FC_OPCODE) != 3);
```

Comment:
The enumeration of `INSN_16` and `INSN_32` can be removed from `src/decode.h`.

Reply:
Add an inline helper function in `src/riscv_private.h` to perform `insn & FC_OPCODE) != 3`.